include/rocksdb/listener.h

```diff
@@ -27,6 +27,7 @@ enum class TableFileCreationReason {
   kFlush,
   kCompaction,
   kRecovery,
+  kMisc,
```

Comment:
It might be good to be more specific with the reason here.

Reply:
Let me rethink this. I was using ```TableFileCreationReason``` to signal ```SstFileManagerImpl::OnAddFile``` that the file was added due to compaction (with kMisc as sort of the default), but it doesn't necessarily mean the file was just created. It could be called during ```DB::Open```, when its just listing all the files. So maybe ```TableFileCreationReason``` is not appropriate. Instead, I can have an explicit parameter in ```OnAddFile``` to specify whether the file being added was created by compaction.